dfa/src/main/scala/dfa/DFA.scala

```diff
@@ -1,3 +1,21 @@
 package dfa // leave this line in the file
 
-// TODO: replace this comment with your implementation
+case class State(label: String)
+case class Transition(from: State, to: State, symbol: Char)
+
+class DFA(val states: Set[State], val transitions: Set[Transition],
+    val start: State, val accept: Set[State]):
+    
+    def accepts(line: String): Boolean = {
+        var currState = start
+        for (c <- line) {
+            // retrieve next state with transition
+            val currTransition = transitions
+                .filter(_.from == currState)
+                .filter(_.symbol == c)
+
+            currState = currTransition.head.to
+        }
```

Comment:
I like the way you wrote this method. It is very readable and simple. That was smart to use filter. I wish I had thought of that.